### ISSUE TITLE
Conflicting package heights error

### ISSUE THREAD

**Issue by bpalermo** [bazelbuild/rules_go#3275]
### What version of rules_go are you using?
v0.29.0

### What version of gazelle are you using?

v0.24.0

### What version of Bazel are you using?

5.2.0

### Does this issue reproduce with the latest releases of all the above?
Yes

### What operating system and processor architecture are you using?
Linux

### Any other potentially useful information about your toolchain?

No

### What did you do?
```
$ bazel build //...
```

### What did you expect to see?
Successful build

### What did you see instead?
```
Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
<unknown line number>: internal compiler error: conflicting package heights 28 and 33 for path "github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
```

Repro: https://github.com/bpalermo/conflicting-package-heights

**Comment by jayconrod** [comment]
"Conflicting package heights" indicates a problem with export data, which is type information for exported definitions in a package. It means that there are multiple versions of an indirect dependency (`github.com/grpc-ecosystem/grpc-gateway/v2/runtime`), and the compiler or linker is detecting a conflict.

Some things to try:

* Upgrade rules_go to the latest version.
* Upgrade Go to the latest version.
* Try to confirm that there are multiple copies of `github.com/grpc-ecosystem/grpc-gateway/v2/runtime`, and if so, ensure only one is part of the build. `bazel query` can be helpful for this.

**Comment by bpalermo** [comment]
Upgraded `go`, `rules_go` and `gazelle`, but the issue is still happening.

When using bazel query I can see both dependencies:
```
bazel query "deps(//pkg/server/...)" | grep grpc_ecosystem_grpc_gateway
Loading: 0 packages loaded
@com_github_grpc_ecosystem_grpc_gateway_v2//internal/httprule:compile.go
...
@grpc_ecosystem_grpc_gateway//:go_apiv2
...
@grpc_ecosystem_grpc_gateway//protoc-gen-grpc-gateway:go_gen_grpc_gateway
@grpc_ecosystem_grpc_gateway//protoc-gen-grpc-gateway:go_gen_grpc_gateway_reset_plugin_
@grpc_ecosystem_grpc_gateway//protoc-gen-grpc-gateway:main.go
@grpc_ecosystem_grpc_gateway//protoc-gen-grpc-gateway:protoc-gen-grpc-gateway
@grpc_ecosystem_grpc_gateway//protoc-gen-grpc-gateway:protoc-gen-grpc-gateway_lib
@grpc_ecosystem_grpc_gateway//protoc-gen-grpc-gateway/internal/gengateway:doc.go
@grpc_ecosystem_grpc_gateway//protoc-gen-grpc-gateway/internal/gengateway:generator.go
@grpc_ecosystem_grpc_gateway//protoc-gen-grpc-gateway/internal/gengateway:gengateway
@grpc_ecosystem_grpc_gateway//protoc-gen-grpc-gateway/internal/gengateway:template.go
@grpc_ecosystem_grpc_gateway//protoc-gen-openapiv2/options:annotations.proto
@grpc_ecosystem_grpc_gateway//protoc-gen-openapiv2/options:openapiv2.proto
@grpc_ecosystem_grpc_gateway//protoc-gen-openapiv2/options:options
@grpc_ecosystem_grpc_gateway//protoc-gen-openapiv2/options:options_go_proto
@grpc_ecosystem_grpc_gateway//protoc-gen-openapiv2/options:options_proto
...
```
I wonder if its due to the compiler `"@grpc_ecosystem_grpc_gateway//protoc-gen-grpc-gateway:go_gen_grpc_gateway"`
```
go_proto_library(
    name = "echo_go_proto",
    compilers = [
        "@io_bazel_rules_go//proto:go_grpc",
        "@grpc_ecosystem_grpc_gateway//protoc-gen-grpc-gateway:go_gen_grpc_gateway",
    ],
    importpath = "github.com/bpalermo/conflicting-package-heights/api/service/echo",
    proto = ":echo_proto",
    visibility = ["//visibility:public"],
    deps = ["@go_googleapis//google/api:annotations_go_proto"],
)
```

**Comment by bpalermo** [comment]
Fixed after bumping `rules_proto_grpc` to  4.2.0 and overriding `go_gen_grpc_gateway`:
```
go_proto_compiler(
    name = "go_gen_grpc_gateway",
    options = [
        "logtostderr=true",
        "allow_repeated_fields_in_body=true",
    ],
    plugin = "@com_github_grpc_ecosystem_grpc_gateway_v2//protoc-gen-grpc-gateway:protoc-gen-grpc-gateway",
    suffix = ".pb.gw.go",
    visibility = ["//visibility:public"],
    deps = [
        "@com_github_grpc_ecosystem_grpc_gateway_v2//runtime:go_default_library",
        "@com_github_grpc_ecosystem_grpc_gateway_v2//utilities:go_default_library",
        "@org_golang_google_grpc//:go_default_library",
        "@org_golang_google_grpc//codes:go_default_library",
        "@org_golang_google_grpc//grpclog:go_default_library",
        "@org_golang_google_grpc//metadata:go_default_library",
        "@org_golang_google_grpc//status:go_default_library",
        "@org_golang_google_protobuf//proto:go_default_library",
    ],
)
```
